validate.go

```diff
@@ -96,6 +97,24 @@ func validateRecurse(ast Expression, fields FieldConfigurations, maxRawValueLeng
 
 				node.Value.Converted = coerced
 			}
+
+			if node.Operator == MatchMatches || node.Operator == MatchNotMatches {
+				var regRaw string
+				if strVal, ok := node.Value.Converted.(string); ok {
+					regRaw = strVal
+				} else if node.Value.Converted == nil {
+					regRaw = node.Value.Raw
+				} else {
+					return 1, fmt.Errorf("Match operator %q cannot be used with fields whose coercion functions return non string values", node.Operator)
+				}
+
+				re, err := regexp.Compile(regRaw)
```

Comment:
Is it worth trying to figure out how to cache the compiled forms?

Reply:
This is sort of caching it. 

The library generally does:

1. parse the expression
2. generate the field configuration (this gets cached already)
3. do a validation pass to ensure the parsed expression is valid for the field configurations and not just valid syntax in general.
4. evaluate

When in being used as part of the `bexpr.Filter` steps 1-3 are done only once.

So the regular expressions get compiled exactly once for the whole lifetime of the `bexpr.Evaluator` that uses it. There is no registry for regular expressions that have ever been used unlike with the field configurations. The logic here is that the types being operated on are likely to be small in number and reused over and over whereas the regexes might not be.